File: src/test/java/com/leo/lab/solution/java/twosum/TwoSumTest.java
package com.leo.lab.solution.java.twosum;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

/**
 * @author Leo Lu
 * @since 2020/05/31
 */
public class TwoSumTest {

    @Before
    public void setUp() {
        // Do nothing
    }

    @Test
    public void testSolution() {
        // GIVEN
        int[] nums = {2, 7, 11, 15};
        int target = 9;

        // WHEN
        Solution solution = new Solution();
        int[] indices = solution.twoSum(nums, target);

        // THEN
        assertArrayEquals(new int[]{0, 1}, indices);
    }
}

File: src/test/kotlin/com/leo/lab/solution/kotlin/twosum/TwoSumTest.kt
package com.leo.lab.solution.kotlin.twosum

import org.junit.Before
import org.junit.Test

import org.junit.Assert.assertArrayEquals

/**
 * @author Leo Lu
 * @since 2020/05/31
 */
class TwoSumTest {

    @Before
    fun setUp() {

    }

    @Test
    fun `test solution`() {
        // GIVEN
        val nums = intArrayOf(2, 7, 11, 15)
        val target = 9

        // WHEN
        val solution = Solution()
        val indices = solution.twoSum(nums, target)

        // THEN
        assertArrayEquals(intArrayOf(0, 1), indices)
    }
}